dbcrossbarlib/src/schema.rs

```diff
@@ -126,9 +126,10 @@ pub enum DataType {
     Int32,
     /// 8-byte integer.
     Int64,
-    /// JSON data. This includes both Postgres `json` and `jsonb` types, the
-    /// differences between which don't usually matter when converting schemas.
+    /// JSON data. This includes Postgre's modern JSONB type, which is always preferred.
     Json,
+    /// JSON in Postgres's old format.
+    OldPgJson,
```

Comment:
Here, you've taken the portable, cross-database `DataType` enum, and added a new `DataType::OldPgJson` that only applies to a single database. Honestly, this feels really dubious: The portable, standardized data type should never include weird database-specific types, and certainly not obsolete ones.

So I think we need to go back to the drawing board, and design a testcase that shows what you want to accomplish. Then from there, we can figure out a clean way to do it, or at least hide any of the necessary hacks in the `postgres` driver itself, and not in the interchange types. Or maybe we should bring back `DataType::Custom(String)`, though I'm less certain of that.

In general, if you _really_ need to produce a PostgreSQL table containing weird things like legacy `json`, `oid`s, or other highly-specialized types, it's a good time to step back to the drawing board.